magefiles/testspecs/templates.go

```diff
@@ -0,0 +1,137 @@
+package testspecs
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"text/template"
+
+	sprig "github.com/go-task/slim-sprig"
+	"github.com/magefile/mage/sh"
+	"k8s.io/klog/v2"
+)
+
+var templates = map[string]string{
+	"test-file":          "templates/test_output_spec.tmpl",
+	"framework-describe": "templates/framework_describe_func.tmpl",
+}
+
+func NewTemplateData(specOutline TestOutline, destination string) *TemplateData {
+
+	re := regexp.MustCompile(`[A-Z][^A-Z]*`)
+	parts := re.FindAllString(specOutline[0].Name, -1)
```

Comment:
For edges cases, like `HASSuiteDescribe`, `E2ESuiteDescribe`, `JVMBuildSuiteDescribe`, and `SPISuiteDescribe`, where it begins with more than one uppercase letter, the regex will return undesired results, I think.

For `SPISuiteDescribe`, for example:

```
func SPISuiteDescribe(text string, args ...interface{}) bool {
	return Describe("[s-p-i-suite "+text+"]", args, Ordered) <--------
}
```

Of course that we can simply edit it.


P.S. I noticed that `HASSuiteDescribe` is not being used. Maybe we can delete it.

Reply:
@rsoaresd Good point! Maybe I can split on `SuiteDescribe` and lower case them separately.